driftbase/match_placements.py

```diff
@@ -0,0 +1,539 @@
+
+        # Verify data integrity
+        player_lobby_id = player_lobby_lock.value
+        if player_lobby_id != lobby_id:
+            log.warning(f"Player '{player_id}' is attempting to start match for lobby '{lobby_id}', but is supposed to be in lobby '{player_lobby_id}'")
+            raise UnauthorizedException(f"You don't have permission to access lobby {lobby_id}")
+
+        with _LockedLobby(_get_lobby_key(lobby_id)) as lobby_lock:
+            lobby = lobby_lock.lobby
+
+            if not lobby:
+                raise RuntimeError(f"Player '{player_id}' is attempting to start match for nonexistent lobby '{lobby_id}'. Player is supposed to be in said lobby")
+
+            # Verify host
+            host_player_id = _get_lobby_host_player_id(lobby)
+            if player_id != host_player_id:
+                log.warning(f"Player '{player_id}' attempted to start the match for lobby '{lobby_id}' without being the lobby host")
+                raise InvalidRequestException(f"You aren't the host of lobby {lobby_id}. Only the lobby host can start the lobby match")
+
+            # Prevent issuing another placement request
+            if lobby["status"] == "starting":
+                log.warning(f"Player '{player_id}' attempted to start the match for lobby '{lobby_id}' while the match is starting")
+                raise InvalidRequestException(f"An active match placement is already in progress for the lobby")
+            
+            # Select match provider
+            match_provider = _get_tenant_config_value("default_match_provider")
+            if match_provider != "gamelift":
+                raise RuntimeError(f"Invalid match provider configured, '{match_provider}'. Only the GameLift match provider is supported at this time")
+
+            # Request a game server
+            lobby_name = lobby["lobby_name"]
+            placement_id = str(uuid.uuid4())
+            max_player_session_count = lobby["team_capacity"] * len(lobby["team_names"])
+            game_session_name = f"Lobby-{lobby_id}-{lobby_name}"
+            custom_data = lobby["custom_data"]
+
+            lobby["placement_id"] = placement_id
+
+            player_latencies = []
+            for member in lobby["members"]:
+                for region, latency in flexmatch.get_player_latency_averages(member["player_id"]).items():
+                    player_latencies.append({
+                        "PlayerId": str(member["player_id"]),
+                        "RegionIdentifier": region,
+                        "LatencyInMilliseconds": latency
+                    })
+
+            log.info(f"Host player '{player_id}' is starting lobby match for lobby '{lobby_id}'. GameLift placement id: '{placement_id}'")
+            response = flexmatch.start_game_session_placement(
+                PlacementId=placement_id,
+                GameSessionQueueName=lobbies._get_tenant_config_value("lobby_game_session_queue"),
```

Comment:
This is something we need to think about; getting the queue from the tenant config means we have to keep the config in sync with the terraform which creates the queues and such, which doesn't sound good to me.  We could ofc decide to use a fixed queue for custom matches, but then that should be codified. Alternatively, somehow discovering or passing in the queue name could be done.

Reply:
agree. we need to solve this better...

Reply:
So the resolution for this is what?